File: DuDoong-Domain/src/main/java/band/gosrock/domain/common/events/host/HostUserJoinEvent.java
package band.gosrock.domain.common.events.host;


import band.gosrock.domain.common.aop.domainEvent.DomainEvent;
import lombok.Builder;
import lombok.Getter;

@Getter
@Builder
public class HostUserJoinEvent extends DomainEvent {
    private final Long hostId;
    private final Long userId;

    public static HostUserJoinEvent of(Long hostId, Long userId) {
        return HostUserJoinEvent.builder().hostId(hostId).userId(userId).build();
    }
}

File: DuDoong-Domain/src/main/kotlin/band/gosrock/domain/common/events/host/HostUserJoinEvent.kt
package band.gosrock.domain.common.events.host

import band.gosrock.domain.common.aop.domainEvent.DomainEvent

class HostUserJoinEvent(
    val hostId: Long?,
    val userId: Long?,
) : DomainEvent() {
    companion object {
        @JvmStatic
        fun of(hostId: Long?, userId: Long?): HostUserJoinEvent =
            HostUserJoinEvent(hostId, userId)
    }

    override fun toString(): String = "HostUserJoinEvent(hostId=$hostId, userId=$userId)"
}
